src/hooks/metamask.hook.ts

```diff
@@ -60,11 +62,64 @@ export function useMetaMask(): MetaMaskInterface {
     return web3.eth.personal.sign(message, address, '');
   }
 
+  async function addContract(address: string, svgData: string): Promise<boolean> {
+    const tokenContract = new web3.eth.Contract(
+      [
+        {
+          constant: true,
+          inputs: [],
+          name: 'symbol',
+          outputs: [
+            {
+              name: '',
+              type: 'string',
+            },
+          ],
+          payable: false,
+          stateMutability: 'view',
+          type: 'function',
+        },
+        {
+          constant: true,
+          inputs: [],
+          name: 'decimals',
+          outputs: [
+            {
+              name: '',
+              type: 'uint8',
+            },
+          ],
+          payable: false,
+          stateMutability: 'view',
+          type: 'function',
+        },
+      ],
+      address,
+    );
+
+    const symbol = await tokenContract.methods.symbol().call();
+    const decimals = await tokenContract.methods.decimals().call();
+
+    return ethereum.sendAsync({
+      method: 'wallet_watchAsset',
+      params: {
+        type: 'ERC20',
+        options: {
+          address,
+          symbol,
+          decimals,
+          image: `data:image/svg+xml;base64,${Buffer.from(svgData).toString('base64')}`,
+        },
+      },
+      id: Math.round(Math.random() * 10000),
```

Comment:
Interesting 😉

Reply:
yeah was quite surprised over this too, but i think it is about having different requests at the same time